sessionctx/variable/session.go

```diff
@@ -448,6 +467,21 @@ func (tc *TransactionContext) RollbackToSavepoint(name string) *SavepointRecord
 	return nil
 }
 
+// FlushStmtPessimisticLockCache merges the current statement pessimistic lock cache into transaction pessimistic lock
+// cache. The caller may need to clear the stmt cache itself.
+func (tc *TransactionContext) FlushStmtPessimisticLockCache() {
+	if tc.CurrentStmtPessimisticLockCache == nil {
+		return
+	}
+	if tc.pessimisticLockCache == nil {
+		tc.pessimisticLockCache = make(map[string][]byte)
```

Comment:
```suggestion
		tc.pessimisticLockCache = tc.CurrentStmtPessimisticLockCache
```
Can we just assign `tc.CurrentStmtPessimisticLockCache` to `tc.pessimisticLockCache` to avoid reallocate?